config.c

```diff
@@ -242,12 +242,17 @@ mdadm_status_t _ident_set_devname(struct mddev_ident *ident, const char *devname
 	else
 		name = devname;
 
-	if (is_name_posix_compatible(name) == false) {
+	if (has_leading_dot(name)) {
```

Comment:
I think it would be worth using the same check order in _ident_set_devname and _ident_set_name
Currently it looks like:
_ident_set_devname:
- has_leading_dot
- is_name_posix_compatible
- is_string_lq

_ident_set_name:
- is_string_lq
- has_leading_dot
- is_name_posix_compatible

Reply:
Right. It is good time to add helper. See updated.